### PR TITLE
Feature/login register pages

### LINKED ISSUE #7: User registration, Authentication, & Authorization
**Is your feature request related to a problem? Please describe.**
Application requires authentication / authorization, account creation is a key part of the MVP.

**Describe the solution you'd like**
- Register page: email, password, password confirm, submit button
- Login page: email, password, submit button
- Form validation (email format, password strength)
- Error messages from backend displayed to user
- Success → redirect to dashboard

**Describe alternatives you've considered**
Considered 'sign up through X', but for the scope of this project it should be a 'nice to have'

**Additional context**
Nothing really!

### PR DESCRIPTION
**Pull request by mpolisch**
## Summary
Implements authentication UI. This includes login, register, and landing pages. Also established frontend project foundation with the tailwind theme, API client, Zod schemas, shared types, and route protection.

## Approach & Design Decisions
- Axios with interceptor: centralizes error normalization so that the backend's shape is normalized onto AxiosError
- Zod schemas on frontend mirror backend, duplicating the validation layer intentionally
- confirmPassword stripped before submission to backend API since it's a frontend only concern
- proxy.ts for route protection: runs before any page renders. Currently only checks if cookie exists, JWT verification coming in next PR
- Suspense on LoginForm: uses UseSearchParams() to read ?registered and ?redirect params, and thus requires Suspense boundary in the app router.
- Tailwind v4 @theme config: designed tokens defined as CSS custom properties in globals.css, allowing for a single source for colours, border radius, etc.

## Security Considerations
- No sensitive data handled in this PR
- Zod client side validation prevents malformed requests from hitting backend
- withCredentials: true ensures cookies are sent on every request, necessary for httpOnly cookie auth
- Open redirect prevention on login redirect param — only relative paths starting with '/' (excluding '//') are accepted, anything else falls back to '/dashboard'
## Related Issues
Closes #7

## Testing
**What's covered:**
- [ ] Unit tests
- [ ] Edge cases (list below)

**Edge cases considered:**
- Already logged in visiting /login or /register: proxy.ts redirects to /dashboard
- Logged out visiting /dashboard: proxy.ts redirects to /login?redirect=/dashboard
- After login, redirect param is kept and user is sent to originally intended route
- Invalid email format caught by Zod before request to backend
- Password mismatch also caught by Zod and error is thrown on confirm password
- Backend errors normalized by axios interceptor, and shown via errors.root
- Network failure: generic fallback message

## Checklist
- [ ] No secrets or credentials in code
- [ ] Input validation in place
- [ ] Error handling covers failure paths
- [ ] Code is self-documenting or commented where non-obvious
- [ ] No dead code or debug logs left in

## Notes
- frontend component testing is not included - requires separate infrastructure setup and can be noted as a future enhancement
- proxy.ts only checks if a cookie exists, not if its valid

### CODE REVIEW

File: proxy.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+
+const AUTH_ROUTES = ['/login', '/register'];
+const PROTECTED_ROUTES = ['/dashboard'];
+
+export function proxy(request: NextRequest) {
+    const { pathname } = request.nextUrl;
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply proxy.ts:8]: outdated suggestion; next.js v16 uses proxy.ts now, and middleware.ts is depracated.

File: .env.example [hunk 2]
```diff
@@ -1 +1 @@
-NEXT_PUBLIC_API_URL=http://localhost:4000
\ No newline at end of file
+NEXT_PUBLIC_API_URL=
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply .env.example]: resolved.

File: lib/api.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+// components can still inspect status codes (401 vs 429 vs 500).
+api.interceptors.response.use(
+    (response) => response,
+    (error) => {
+        if (axios.isAxiosError(error) && error.response?.data?.message) {
+            error.message = error.response.data.message;
+        }
+        return Promise.reject(error);
+    },
+);
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply lib/api.ts:23]: way over engineered, this is wild
